pkg/clockcache/cache.go

```diff
@@ -45,11 +52,11 @@ func WithMax(max uint64) *Cache {
 // Insert a key/value mapping into the cache if the key is not already present
 // returns the canonical version of the value
 // and if the value is in the map
-func (self *Cache) Insert(key interface{}, value interface{}) (canonicalValue interface{}, in_cache bool) {
+func (self *Cache) Insert(key interface{}, value interface{}, size uint64) (canonicalValue interface{}, in_cache bool) {
```

Comment:
This and `InsertBatch` need updates to the comments to include the new argument.

~~Also, would it be a good idea to create convenience methods that calculate the sizes themselves? Manually setting the size every time is a bit error-prone, could lead to subtle mistakes.~~

On second thought, I guess it should be fine since we don't really use this directly anyways.

Reply:
One stable way i know of is converting anything to a byte slice. For example:

https://golang.org/pkg/encoding/binary/#Write
```
buf := new(bytes.Buffer)
var num int = 1234
err := binary.Write(buf, binary.LittleEndian, num)
if err != nil {
    fmt.Println("binary.Write failed:", err)
}
```

But I am not sure if that's production recommended.

Reply:
Added comments and changed variable name to make it clear this is bytes. We are not doing the binary encoding since that is very expensive